components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,394 @@
+  cell.action = _actions[indexPath.row];
+  cell.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
+  cell.backgroundColor = self.backgroundColor;
+  cell.actionFont = self.actionFont;
+  return cell;
+}
+
+- (void)setTitle:(NSString *)title {
+  _header.title = title;
+  [self setInvalidPreferredContentSize];
+}
+
+- (NSString *)title {
+  return _header.title;
+}
+
+- (void)setMessage:(NSString *)message {
+  _header.message = message;
+  [self setInvalidPreferredContentSize];
+}
+
+- (NSString *)message {
+  return _header.message;
+}
+
+- (void)setTitleFont:(UIFont *)titleFont {
+  _header.titleFont = titleFont;
+}
+
+- (UIFont *)titleFont {
+  return _header.titleFont;
+}
+
+- (void)setMessageFont:(UIFont *)messageFont {
+  _header.messageFont = messageFont;
+}
+
+- (UIFont *)messageFont {
+  return _header.messageFont;
+}
+
+- (void)setBackgroundColor:(UIColor *)backgroundColor {
+  self.view.backgroundColor = backgroundColor;
+  _tableView.backgroundColor = backgroundColor;
+  [_tableView setNeedsLayout];
+  _header.backgroundColor = backgroundColor;
+}
+
+- (UIColor *)backgroundColor {
+  return self.view.backgroundColor;
+}
+
+#pragma mark - Dynamic Type
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+  _header.mdc_adjustsFontForContentSizeCategory = adjusts;
+  [self updateFontsForDynamicType];
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(updateFontsForDynamicType)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  } else {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UIContentSizeCategoryDidChangeNotification
+                                                  object:nil];
+  }
+  [self.view setNeedsLayout];
+}
+
++ (UIFont *)actionFontDefault {
+  if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+    return [UIFont mdc_standardFontForMaterialTextStyle:MDCFontTextStyleSubheadline];
+  }
+  return [MDCTypography subheadFont];
+}
+
+- (void)updateTableFonts {
+  UIFont *finalActionsFont = _actionFont ?: [[self class] actionFontDefault];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    finalActionsFont =
+        [finalActionsFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleSubheadline
+                                       scaledForDynamicType:self.mdc_adjustsFontForContentSizeCategory];
+  }
+  _actionFont = finalActionsFont;
+  [self updateTable];
+}
+
+- (void)updateFontsForDynamicType {
+  [self updateTableFonts];
+  [self setInvalidPreferredContentSize];
+}
+
+- (void)setInvalidPreferredContentSize {
+  _invalidPreferredContentSize = YES;
```

Comment:
It looks like we might be able to drop this ivar altogether now and just rely on setNeedsLayout. Is that possible?

Reply:
We need it because viewWillTransitionToSize requires it.